server/src/com/cloud/network/NetworkModelImpl.java

```diff
@@ -1637,7 +1637,8 @@ public void checkNetworkPermissions(Account owner, Network network) {
             throw new CloudRuntimeException("cannot check permissions on (Network) <null>");
         }
         // Perform account permission check
-        if (network.getGuestType() != Network.GuestType.Shared || (network.getGuestType() == Network.GuestType.Shared && network.getAclType() == ACLType.Account)) {
+        if ((network.getGuestType() != Network.GuestType.Shared && network.getGuestType() != GuestType.L2) ||
```

Comment:
Network.GuestType.Shared is used and then GuestType.L2 directly. please cleanup the imports and use only one scope for one java class.